.github/workflows/check-clean-branch.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+# Exits if any uncommitted changes are found.
+
+clean=$(git status | grep "nothing to commit, working tree clean")
```

Comment:
can use `git status --porcelain` empty output to indicate clean working tree instead of parsing the output text

Reply:
Actually looks like `dit diff --quiet --exitcode` also works.

```
       --exit-code
           Make the program exit with codes similar to diff(1). That is, it exits with 1 if there were differences and 0 means no differences.
```

Reply:
Thanks for the link- parsing the output text is really jank - will update